Sources/ForTag.swift

```diff
@@ -124,25 +131,115 @@ class ForNode : NodeType {
 
     if !values.isEmpty {
       let count = values.count
+      var result = ""
+
+      var currentLoopContext = context["forloop"] as? [String: Any]
```

Comment:
I was a bit confused on how you were getting the previously existing loop contexts, maybe give this a more descriptive name? For example `existingLoopContext` or `parentLoopContext`?